### PR TITLE
Enable nullability in some GridEntry members

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in some GridEntry members.

###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/10832)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/Controls/PropertyGrid/PropertyGridInternal/GridEntry.cs [hunk 1]
```diff
@@ -849,11 +849,11 @@ internal void ClearCachedValues(bool clearChildren = true)
     /// <summary>
     ///  Converts the given string of text to a value.
     /// </summary>
-    public object ConvertTextToValue(string text)
+    public object? ConvertTextToValue(string? text)
     {
         if (TypeConverter.CanConvertFrom(this, typeof(string)))
         {
-            return TypeConverter.ConvertFromString(this, text);
+            return TypeConverter.ConvertFromString(this, text!);
```
Comment (lonitra) [inline src/System.Windows.Forms/src/System/Windows/Forms/Controls/PropertyGrid/PropertyGridInternal/GridEntry.cs:856]: Do we get `NotSupportedException` thrown when `text = null`? If so, I don't think we should be indicating that null is an acceptable parameter for this method

Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/Controls/PropertyGrid/PropertyGridInternal/GridEntry.cs:856]: This seems to return an empty string when text is null.

Comment (lonitra) [reply src/System.Windows.Forms/src/System/Windows/Forms/Controls/PropertyGrid/PropertyGridInternal/GridEntry.cs:856]: Thanks for checking that! Could you add a comment that we will get empty string when text = null for others looking at this next time?

**Review by lonitra — APPROVED** [review]
🚀

**Pull request merged**